src/helpers.js

```diff
@@ -40,4 +40,36 @@ export const cloneObject = (object) => {
         results[key] = data;
     });
     return results;
+};
+
+export const hasDiffrentKeysLength = (srcObject, compareObject) => {
+    return Object.keys(srcObject).length !== Object.keys(compareObject).length;
+};
+
+export const isNotEqualArray = (srcArray, compareArray) => {
+    if (srcArray.length !== compareArray.length) return true;
+    let result = false;
+    srcArray.forEach((value, index) => {
+        if (isNotEqualValue(compareArray[index], value)) result = true;
+    });
+    return result;
+};
+
+export const isNotEqualValue = (srcValue, compareValue) => {
+    const srcPropType = typeof srcValue;
+    const comparePropType = typeof compareValue;
+    if (srcPropType !== comparePropType || (srcPropType !== 'object') && srcValue !== compareValue) return true;
+    if (Array.isArray(srcValue)) {
+        return isNotEqualArray(srcValue, compareValue);
+    }
+    if (srcPropType === 'object' && isNotEqualObject(srcValue, compareValue)) return true;
+};
+
+export const isNotEqualObject = (srcObject, compareObject) => {
+    if (hasDiffrentKeysLength(srcObject, compareObject)) return true;
+    let result = false;
+    Object.keys(srcObject).forEach((key) => {
```

Comment:
[Could] Instead of using `forEach` and overwriting the `result` variable, you could filter out keys that do not match your condition.

```javascript
return Object.keys(srcObject).filter(key => isNotEqualValue(srcObject[key], compareObject[key])).length > 1
```